citro3d/src/buffers.rs

```diff
@@ -0,0 +1,101 @@
+use std::marker::PhantomData;
+use std::ops::{Deref, DerefMut};
+use std::sync::{LazyLock, RwLock};
+
+use crate::attrib;
+
+static BUF_INFO: LazyLock<RwLock<BufInfo>> = LazyLock::new(|| {
+    let raw = unsafe {
+        let info = citro3d_sys::C3D_GetBufInfo();
+        citro3d_sys::BufInfo_Init(info);
+        info
+    };
+
+    RwLock::new(BufInfo { raw })
+});
+
+pub struct BufInfo {
+    raw: *mut citro3d_sys::C3D_BufInfo,
+}
+
+// SAFETY: the RWLock ensures unique access when mutating the global struct, and
+// we trust citro3d to Do The Right Thing™ and not mutate it otherwise.
+unsafe impl Sync for BufInfo {}
+unsafe impl Send for BufInfo {}
+
+// TODO: is this a good name? It's more like a "handle" to the VBO data, or a slice.
+#[derive(Debug, Clone, Copy)]
+pub struct Index<'vbo> {
```

Comment:
This could maybe be `BufSlice`?

Reply:
I decided to just call it `Slice` so you can use it as `buffer::Slice`.